src/wpantund/NCPInstanceBase.cpp

```diff
@@ -329,6 +337,13 @@ NCPInstanceBase::property_get_value(
 	} else if (strcaseequal(key.c_str(), kWPANTUNDProperty_DaemonEnabled)) {
 		cb(0, boost::any(mEnabled));
 
+	} else if (strcaseequal(key.c_str(), kWPANTUNDProperty_NCPRole)) {
+		std::string role = kWPANTUNDRole_Detached;
+		if (mEnabled && ncp_state_is_associated(get_ncp_state())) {
+			role = kWPANTUNDRole_EndDevice;
```

Comment:
Current code, we seem to return `EndDevice` or `Detached`. Is it intended?
How about getting the `SPINEL_PROP_NET_ROLE` and mapping it to string?